video_xblock/static/js/transcripts-autoload.js

```diff
@@ -0,0 +1,248 @@
+/**
+ * Default transcripts functionality is represented here.
+ */
+
+/** Run a callback when DOM is fully loaded */
+var domReady = function(callback) {
+    if (document.readyState === "interactive" || document.readyState === "complete") {
+        callback();
+    } else {
+        document.addEventListener("DOMContentLoaded", callback);
+    }
+};
+
+/** Get url of a specific transcript from a given transcripts array. */
+var getTranscriptUrl = function(transcriptsArray, langCode){
+    var url = '';
+    transcriptsArray.forEach(function(sub){
+        if(sub['lang']==langCode){
+          url = sub['url'];
+        }
+    });
+    return url;
+};
+
+/** Create elements to display messages with status on actions with default transcripts. */
+var createStatusMessageElement = function(langCode, actionSelector){
```

Comment:
Let's define these using function declarations instead of function expressions http://stackoverflow.com/questions/336859/javascript-function-declaration-syntax-var-fn-function-vs-function-fn to take advantage of hoisting for this and other functions.